redis/commands/search/hybrid_query.py

```diff
@@ -0,0 +1,326 @@
+        self._scorer = scorer
+        self._yield_score_as = yield_score_as
+
+    def query_string(self) -> str:
+        """Return the query string of this query object."""
+        return self._query_string
+
+    def scorer(self, scorer: str) -> "HybridSearchQuery":
+        """
+        Scoring algorithm for text search query.
+        Allowed values are "TFIDF" or  "BM25"
+        """
+        self._scorer = scorer
+        return self
+
+    def yield_score_as(self, alias: str) -> "HybridSearchQuery":
+        """
+        Yield the score as a field.
+        """
+        self._yield_score_as = alias
+        return self
+
+    def get_args(self) -> List[str]:
+        args = ["SEARCH", self._query_string]
+        if self._scorer:
+            args.extend(("SCORER", self._scorer))
+        if self._yield_score_as:
+            args.extend(("YIELD_SCORE_AS", self._yield_score_as))
+        return args
+
+
+class HybridVsimQuery:
+    def __init__(
+        self,
+        vector_field_name: str,
+        vector_data: Union[bytes, str],
+        vsim_search_method: Optional[str] = None,
+        vsim_search_method_params: Optional[Dict[str, Any]] = None,
+        filter: Optional["Filter"] = None,
+    ) -> None:
+        """
+        Create a new hybrid vsim query object.
+
+        Args:
+            vector_field_name: Vector field name.
+            vector_data: Vector data for the search.
+            vsim_search_method: Search method that will be used for the vsim search.
+                Allowed values are "KNN" or "RANGE".
+            vsim_search_method_params: Search method parameters. Use the param names
+                for keys and the values for the values. Example: {"K": 10, "EF_RUNTIME": 100}.
+            filter: If defined, a filter will be applied on the vsim query results.
+        """
+        self._vector_field = vector_field_name
+        self._vector_data = vector_data
+        if vsim_search_method and vsim_search_method_params:
+            self.vsim_method_params(vsim_search_method, **vsim_search_method_params)
+        else:
+            self._vsim_method_params = None
+        self._filter = filter
+
+    def vector_field(self) -> str:
+        """Return the vector field name of this query object."""
+        return self._vector_field
+
+    def vector_data(self) -> Union[bytes, str]:
+        """Return the vector data of this query object."""
+        return self._vector_data
+
+    def vsim_method_params(
+        self,
+        method: str,
+        **kwargs,
+    ) -> "HybridVsimQuery":
+        """
+        Add search method parameters to the query.
+
+        Args:
+            method: Vector search method name. Supported values are "KNN" or "RANGE".
+            kwargs: Search method parameters. Use the param names for keys and the
+                values for the values. Example: {"K": 10, "EF_RUNTIME": 100}.
+        """
+        vsim_method_params: List[Union[str, int]] = [method]
```

Comment:
It's not really clear which parameters are corresponding to which method and which one is optional, which one is required. I would suggest to add more clear structures around it for better user experience